scripts/bert/run_pretraining.py

```diff
@@ -31,81 +31,170 @@
+from pretraining_utils import split_and_load, log, evaluate, get_argparser
 from pretraining_utils import save_parameters, save_states, profile
+from pretraining_utils import get_pretrain_data_text, generate_dev_set
 
-# arg parser
+# parser
 parser = get_argparser()
-parser.add_argument('--gpus', type=str, default='0', help='List of GPUs to use. e.g. 1,3')
-parser.add_argument('--kvstore', type=str, default='device', help='KVStore type')
-args = parser.parse_args()
+parser.add_argument('--raw', action='store_true',
+                    help='If set, both training and dev samples are generated on-the-fly '
+                         'from raw texts instead of pre-processed npz files. ')
+parser.add_argument('--max_seq_length', type=int, default=512,
+                    help='Maximum input sequence length. Effective only if --raw is set.')
+parser.add_argument('--short_seq_prob', type=float, default=0.1,
+                    help='The probability of producing sequences shorter than max_seq_length. '
+                         'Effective only if --raw is set.')
+parser.add_argument('--masked_lm_prob', type=float, default=0.15,
+                    help='Probability for masks. Effective only if --raw is set.')
+parser.add_argument('--max_predictions_per_seq', type=int, default=80,
+                    help='Maximum number of predictions per sequence. '
+                         'Effective only if --raw is set.')
+parser.add_argument('--cased', action='store_true',
+                    help='Whether to tokenize with cased characters. '
+                         'Effective only if --raw is set.')
+parser.add_argument('--whole_word_mask', action='store_true',
+                    help='Whether to use whole word masking rather than per-subword masking.'
+                         'Effective only if --raw is set.')
+parser.add_argument('--sentencepiece', default=None, type=str,
+                    help='Path to the sentencepiece .model file for both tokenization and vocab. '
+                         'Effective only if --raw is set.')
+parser.add_argument('--sp_nbest', type=int, default=0,
+                    help='Number of best candidates for sampling subwords with sentencepiece. '
+                         'Effective only if --raw is set.')
+parser.add_argument('--sp_alpha', type=float, default=1.0,
+                    help='Inverse temperature for probability rescaling for sentencepiece '
+                         'sampling. Effective only if --raw is set.')
+parser.add_argument('--num_data_workers', type=int, default=8,
+                    help='Number of workers to pre-process data. '
+                         'Effective only if --raw is set.')
+parser.add_argument('--eval_use_npz', action='store_true',
+                    help='Set to True if --data_eval provides npz files instead of raw text files')
+parser.add_argument('--backend', type=str, default='device',
```

Comment:
The naming --backend seems not very clear to user. Maybe rename it to "comm_backend"?